classes/task/purge_profile_groups.php

```diff
@@ -0,0 +1,47 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_excimer\task;
+
+/**
+ * Purge old profile group data.
+ *
+ * @package   tool_excimer
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class purge_profile_groups extends \core\task\scheduled_task {
+
+    /**
+     * Get a descriptive name for this task (shown to admins).
+     *
+     * @return string
+     */
+    public function get_name() {
+        return get_string('task_purge_profile_groups', 'tool_excimer');
+    }
+
+    /**
+     * Do the job.
+     */
+    public function execute() {
+        global $DB;
+
+        $month = userdate(date('2 months ago'), '%Y%m');
```

Comment:
Seeing as you've implemented this as a full history I'm inclined to have the option to keep the data, so can you make this an admin option and make it purge data older than 13 months by default which means at any give time with the partial month in progress we have a full years worth of data. Maybe make the setting say 12 months but add the +1 in code

Reply:
done